### PR TITLE
Ensure VMI deletion before proceeding to other deletes

### PR DESCRIPTION
**Pull request by akalenyu**
Signed-off-by: Alex Kalenyuk <akalenyu@redhat.com>

**What this PR does / why we need it**:
In the imageupload tests, we start looping over the PVC being gone before we know for fact the VMI is.
Let's ensure VMI deletion so we only proceed to PVC/others after that succeeds, avoiding a timeout.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/storage/imageupload.go
Comment (rmohr) [inline tests/storage/imageupload.go:185]: In general tests should not have to wait for a previous test to be fully cleaned up.
Normally it works like this right now:
 * In the outmost `BeforeEach`, you just call `tests.BeforeTestCleanup`. That will delete everything in the test namespace, but it will not wait for artifacts to disappear since that can take quite some time
 * You then simply create and operate on entities with random names and do your checks. The clusters have a size which allows to run at least three fedora VMs (they are the biggest VMs we use) in parallel without impacting each other.
 * Specific artifact cleanup and waiting for resources to disappear is therefore normally not necessary at all.

**Review by rmohr — CHANGES_REQUESTED** [review]

File: tests/storage/imageupload.go
Comment (akalenyu) [reply tests/storage/imageupload.go:185]: I see, thanks for the elaborate explanation

Just to clarify the situation here - this test ensures DV cleanup, which in turn also ensures PVC deletion.
The problem is (at least that is what I see in [periodic](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-1.20-sig-storage/1437043412187484160)) that we start waiting (90 sec) on the PVC being gone before the VMI is, losing some valuable seconds and then timing out.

Comment (rmohr) [reply tests/storage/imageupload.go:185]: Thanks for the details. The latest state of the PR seems to do something completely different now though :)

Comment (akalenyu) [reply tests/storage/imageupload.go:185]: I dropped the VMI deletion in snapshot/restore tests, that's all
Just want to start looping over PVCs being gone after we know VMI's are, to not lose any of the 90sec.

**Comment by Gre8t** [comment]
hi @akalenyu i just went through this conversation and noticed that you need to add the label "lgtm and approved" before @rmohr can approve the merging of your pr.

**Comment by akalenyu** [comment]
We stopped seeing this specific issue in 1.21 CI lanes and onwards, let's close.

**Pull request closed without merging**
